app/services/payments/save_payment_service.rb

```diff
@@ -0,0 +1,14 @@
+module Payments
+  class SavePaymentService < ApplicationService
+    # @attr_reader params [Hash]
+    # - token: [String] Current user token
+    # - pm_id: [String] Payment id
+    # - type: [String] Payment type
+
+
+    def call
+      user = Users::CurrentUserService.perform(@params[:token])
```

Comment:
you should pass `user` to service throu `params` hash instead of fetching from DB